### DISCUSSION TITLE
upload

### DISCUSSION THREAD

**Discussion by Tamilvip07** [tulir/whatsmeow#254] [WhatsApp protocol Q&A]
hi i am kinda a newbie with golang is there any example to upload a file

**Comment by Tamilvip07** [comment]
too many arguments in call to c.SendMessage
	have ("context".Context, string, string, *"go.mau.fi/whatsmeow/binary/proto".Message)
	want (types.JID, string, *"go.mau.fi/whatsmeow/binary/proto".Message)compiler[WrongArgCount](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal?utm_source%3Dgopls#WrongArgCount)
	
	
	
my code:
`
uploaded, err := c.Upload(ct.Background(), file, whatsmeow.MediaDocument)
	msg := &waProto.DocumentMessage{
		Mimetype:      proto.String("text/plain"),
		Url:           &uploaded.URL,
		Title:         proto.String(fileName),
		FileSha256:    uploaded.FileEncSHA256,
		FileLength:    &uploaded.FileLength,
		FileEncSha256: uploaded.FileEncSHA256,
		MediaKey:      uploaded.MediaKey,
		DirectPath:    &uploaded.DirectPath,
	}
_, _ = c.SendMessage(ct.Background(), ctx.Message.Info.Chat.String(), "", &waProto.Message{
		DocumentMessage: msg,
	})`

**Comment by mamur-rezeki** [comment]
You need to update the `whatsmeow` package to the latest.

**Reply by IndrajeethY** [reply]
hello there! I have updated my whatsmeow version https://nekobin.com/juruloqedo this is my code the issue is 
![image](https://user-images.githubusercontent.com/79161058/198339365-fa00972a-be81-479a-a530-00bfe613105e.png)
i am gettin invalid file. Can i know why is this happeneing>

**Reply by mamur-rezeki** [reply]
What's mean of invalid file ?

Have you set the `Mimetype` field.

In some cases, different mimetype would be a problem while try to open the file directly from mobile app.

i.e. PDF :
`application/pdf` vs `application/wps-office.pdf`

**Reply by IndrajeethY** [reply]
> What's mean of invalid file ?
> 
> Have you set the `Mimetype` field.
> 
> In some cases, different mimetype would be a problem while try to open the file directly from mobile app.
> 
> i.e. PDF : `application/pdf` vs `application/wps-office.pdf`

Yeh I have mentioned mimetype you could check the codes which i have linked

**Reply by IndrajeethY** [reply]
> > What's mean of invalid file ?
> > Have you set the `Mimetype` field.
> > In some cases, different mimetype would be a problem while try to open the file directly from mobile app.
> > i.e. PDF : `application/pdf` vs `application/wps-office.pdf`
> 
> Yeh I have mentioned mimetype you could check the codes which i have linked

Also it's even in valid while trying to open it in WhatsApp web version in pc

**Reply by IndrajeethY** [reply]
nvm anyway sorted out damn i was blind

**Reply by mamur-rezeki** [reply]
Have you set the default program to open "plain/text" file ?

I can't see the problem of your codes.
